collect_app/src/test/java/org/odk/collect/android/utilities/MultiFormDownloaderTest.java

```diff
@@ -1,414 +0,0 @@
-                "                </data>\n" +
-                "            </instance>\n" +
-                "            <instance id=\"__last-saved\" src=\"jr://instance/last-saved\"/>\n" +
-                "            <bind nodeset=\"/data/q1\" type=\"string\"/>\n" +
-                "           <setvalue event=\"odk-instance-first-load\" ref=\"/data/q1\" value=\" instance('__last-saved')/data/q1 \"/>\n" +
-                "        </model>\n" +
-                "    </h:head>\n" +
-                "    <h:body>\n" +
-                "        <input ref=\"/data/q1\">\n" +
-                "            <label>Question</label>\n" +
-                "        </input>\n" +
-                "    </h:body>\n" +
-                "</h:html>";
-        File formXml = File.createTempFile("basicLastSaved", ".xml");
-        formXml.deleteOnExit();
-
-        BufferedWriter out = new BufferedWriter(new FileWriter(formXml));
-        out.write(basicLastSaved);
-        out.close();
-
-        MultiFormDownloader downloader = spy(new MultiFormDownloader(openRosaXMLFetcher));
-        ServerFormDetails test1 = new ServerFormDetails("Last Saved", "https://testserver/media.xml",
-                "https://testserver/media-manifest.xml", "basic-last-saved", "20200101",
-                "hash", "manifestHash", false, false);
-        MultiFormDownloader.FileResult result = new MultiFormDownloader.FileResult(formXml, true);
-        doReturn(result).when(downloader).downloadXform(test1.getFormName(), test1.getDownloadUrl(), null);
-        doReturn("").when(downloader).downloadManifestAndMediaFiles(any(), any(), any(), anyInt(), anyInt(), any());
-        doReturn(true).when(downloader).installEverything(any(), any(), any());
-
-        List<ServerFormDetails> forms = new ArrayList<>();
-        forms.add(test1);
-
-        HashMap<ServerFormDetails, String> messages = downloader.downloadForms(forms, null);
-        assertThat(messages.get(test1), is("Success"));
-    }
-
-    /**
-     * Edge case: a form could have an attachment with filename last-saved.xml. This will get
-     * replaced immediately on download and this test documents that behavior. We could let it go
-     * through but let's replace it immediately to help a user who tries this troubleshoot.
-     * Otherwise it would only be replaced when an instance is saved so a user could think everything
-     * is ok if they only try launching the form once.
-     *
-     * This is an unfortunate side effect of using the form media folder to store the contents that
-     * jr://instance/last-saved resolves to.
-     *
-     * Additionally, immediately replacing a secondary instance with name last-saved.xml avoid users
-     * exploiting this current implementation quirk as a feature to preload defaults for the first
-     * instance.
-     * */
-    @Test
-    public void downloadingFormWithExternalSecondaryInstanceNamedLastSavedXml_Succeeds() throws Exception {
```

Comment:
I think this edge case is also important to capture somewhere.

Reply:
Same as the media file progress test I think. Didn't need it but I think you're right we should add it in any way.

Reply:
The comment is the most important part, I think. But I think it's also important to have a test around which `last-saved` external secondary instance gets used in that specific case.